### DISCUSSION TITLE
Property List and waypoint navigation

### DISCUSSION THREAD

**Discussion by harryl2021** [JSBSim-Team/jsbsim#408] [General]
I'm new to jsbsim and trying to understand it. I want to know is there a way to list all the properties that jsbsim make use of such as `position/lat-rad, position /long-rad` etc. Also I want to know if there is no autopilot file for some aircraft then how do I specify waypoints so that the aircraft travels along those waypoints. I tried to copy the C310ap for 737 aircraft(same for script and initialization files of the c310 aircraft  by making the changes to use aircraft property element etc. in 737wp_.xml script file). But this is not working as the aircraft is not at all moving from its initial position.

**Comment by bcoconni** [comment]
Hi @harryl2021 and welcome to JSBSim ! 
>  I want to know is there a way to list all the properties that jsbsim make use of such as `position/lat-rad`, `position /long-rad` etc.

Sure. If you are using the executable `JSBSim.exe`, you can specify the `--catalog` option to the command line interface in your terminal. For example, the command line below will display all the properties of the `737` aircraft:
```bash
> JSBSim.exe --aircraft=737 --catalog
```
If you are using python code, you should use one of the methods `query_property_catalog` or `get_property_catalog`. If you are using C++ code then you should use the method `JSBSim::FGFDMExec::QueryPropertyCatalog`.
> I tried to copy the C310ap for 737 aircraft(same for script and initialization files of the c310 aircraft by making the changes to use aircraft property element etc. in 737wp_.xml script file). But this is not working as the aircraft is not at all moving from its initial position.

You must tell JSBSim which initialization file you are using: 

#### Option 1 - Using the command line interface
You can supply the option`initfile=` to `JSBSim.exe`. For example, you can tell JSBSim to use the initialization file `cruise_init.xml` with the  command line below. Note that, for this command to succeed, the file `cruise_init.xml` must be located in the same folder than the `737` aircraft.
```bash
> JSBSim.exe --aircraft=737 --initfile=cruise_init.xml
```
#### Option 2 - Updating the script file
Alternatively you can modify the `initfile` attribute of the `<use>` tag in your script file:
https://github.com/JSBSim-Team/jsbsim/blob/375f5be0fdfeb210f4cf66f938e3e6fd426c2966/scripts/737_cruise.xml#L13
Here again, the file `cruise_init.xml` must be located in the same folder than the `737`
